### DISCUSSION TITLE
How to start reading books imported from OPDS feeds

### DISCUSSION THREAD

**Discussion by cary-rowen** [edrlab/thorium-reader#1893] [Q&A]
I added Gutenberg, Then I found The Complete Works of Shakespeare,
Click "Import epub" in the menu,
How do I start reading after the download is complete, I can't even find the books I've downloaded.

The OPDS I added is as follows:
 http://m.gutenberg.org/ebooks.opds/

**Comment by cary-rowen** [comment]
Oh, I found out, I seem to have to restart Thorium

**Comment by gautierchomel** [comment]
Once Imported the book will appear in the My books tab. Usually restarting Thorium is not needed.

![image](https://user-images.githubusercontent.com/107039498/204521986-fc929336-39cc-4e5c-8867-6512f48a43ae.png)

**Comment by danielweck** [comment]
The UX (User Experience) is indeed a bit confusing, because once an OPDS / catalog import is finished, there is no automatic GUI (Graphical User Interface) redirection into the local bookshelf. The UX design could be improved by immediately opening the publication library (EPUB, PDF, audiobooks will all initially appear in the top "recently added" horizontal strip) once the download is finished, but this could in fact make the UX worse when several publications are downloaded in parallel (for example, large audiobooks will take longer to import than small ebooks). Maybe Thorium could detect that no other download is pending, and the GUI could switch automatically into the local bookshelf. Personally I think this would introduce unnecessary code complexity and I am sure there will be UX edge cases resulting in bad user feedback. Perhaps a better solution would be to display a prominent temporary GUI button that the user could click in order to redirect the keyboard and visual focus on the "recently added" section of the local bookshelf. This would be accessible, user-friendly, and safe.

**Reply by cary-rowen** [reply]
Thank you for clarifying this.

**Reply by gautierchomel** [reply]
I think about two little things to prettify UX : 
* Once a book has been imported mention that in the OPDS catalog (maybe a tag "imported" could do the job)
* Propose a link to the imported book once it is downloaded (maybe in the alert box that says "The import of ... is finished")

Also having an option to setup a tag for books imported from OPDS could help retrieve them in the bookshelf. It could be generic (imported from OPDS) or get the OPDS feed name (but it would be bad experience if the OPDS name is long as some of us use the OPDS address as Name).

**Comment by cary-rowen** [comment]
oh, thanks!

在 2022/11/29 19:57, Gautier Chomel 写道:
